partition/grub.go

```diff
@@ -63,28 +63,41 @@ func (g *grub) envFile() string {
 	return filepath.Join(g.Dir(), "grubenv")
 }
 
-func (g *grub) GetBootVar(name string) (string, error) {
+func (g *grub) GetBootVars(names []string) (map[string]string, error) {
+	out := map[string]string{}
+
 	// Grub doesn't provide a get verb, so retrieve all values and
 	// search for the required variable ourselves.
 	output, err := runCommand(grubEnvCmd, g.envFile(), "list")
 	if err != nil {
-		return "", err
+		return nil, err
 	}
 
 	cfg := goconfigparser.New()
 	cfg.AllowNoSectionHeader = true
 	if err := cfg.ReadString(output); err != nil {
-		return "", err
+		return nil, err
+	}
+
+	for _, name := range names {
+		v, err := cfg.Get("", name)
+		if err != nil {
+			return nil, err
+		}
+		out[name] = v
 	}
 
-	return cfg.Get("", name)
+	return out, nil
 }
 
-func (g *grub) SetBootVar(name, value string) error {
+func (g *grub) SetBootVars(values map[string]string) error {
 	// note that strings are not quoted since because
 	// RunCommand() does not use a shell and thus adding quotes
 	// stores them in the environment file (which is not desirable)
-	arg := fmt.Sprintf("%s=%s", name, value)
+	arg := ""
+	for k, v := range values {
+		arg += fmt.Sprintf("%s=%s ", k, v)
```

Comment:
Given the comment above, this doesn't look right as it'll shove all the values in a single argument. The end result of `["set", "foo=1 bar=2"]` might well be `"foo"` being set to `"1 bar=2"`.